pkg/controller/control.go

```diff
@@ -1287,3 +1329,79 @@ func (c *Controller) GetLatestMetics() *enginerpc.Metrics {
+	// - If it does not, we likely can't do a namespace swap anyway, since we don't have access to /host/proc.
+	// - If it does, we just need to know where in the container we can access the mount points to create a bind mount.
+	sourcePoints, err := mounter.List()
+	if err != nil {
+		log.WithError(err).Warn("Failed to list mount points while deciding whether or not to freeze")
+	}
+
+	for _, sourcePoint := range sourcePoints {
+		if sourcePoint.Device != devicePath {
+			continue // We are iterating through the list of all mounted file systems.
+		}
+		log.Debugf("Mounting file system to %v", freezePoint)
+		if err = attemptBindMountFileSystem(sourcePoint.Path, freezePoint, mounter); err != nil {
+			// There's no particular reason to think we will be successful with a different sourcePoint.
+			err = errors.Wrapf(err, "failed to bind mount file system from %v to %v", sourcePoint, freezePoint)
+			return
+		}
+		mounted = true
+
+		// We must verify it is still save to freeze the file system. If the source mount was unmounted by someone else
+		// before we bind mounted, the bind mount could refer to the root file system.
+		var device string
+		device, _, err = mount.GetDeviceNameFromMount(mounter, freezePoint)
+		if err != nil || device != devicePath {
+			// This would likely only happen if an upper layer unmounted the volume. If that is the case, we are
+			// probably being torn down. Trying other sourcePoints will likely lead to a pointless race.
+			err = errors.Wrapf(err, "cannot verify mount point %v refers to volume", freezePoint)
+			return
+		}
+
+		log.Infof("Freezing file system mounted at %v", freezePoint)
+		if err = util.AttemptFreezeFileSystem(freezePoint, exec); err == nil {
+			frozen = true
+		}
+		break
+	}
+
+	return
+}
+
+// attemptBindMountFileSystem attempts to bind mount sourcePoint to mountPoint. attemptBindMountFileSystem considers it
+// an error if mountPoint already has a file system mounted. This likely indicates another freeze is in progress. We do
+// not want to double mount, etc.
+func attemptBindMountFileSystem(sourcePoint, mountPoint string, mounter mount.Interface) error {
+	if err := os.MkdirAll(mountPoint, 0700); err != nil {
+		return err
+	}
+
+	isMountPoint, err := mounter.IsMountPoint(mountPoint)
+	if err != nil {
+		return err
+	}
+	if isMountPoint {
+		return errors.Wrapf(err, "file system is already mounted to %v", mountPoint)
+	}
+
+	return mounter.Mount(sourcePoint, mountPoint, "", []string{"bind"})
+}
+
+// attemptUnmountFileSystem attempts to unmount the file system mounted at mountPoint. Returns success or failure.
+func attemptUnmountFileSystem(mountPoint string, mounter mount.Interface, log logrus.FieldLogger) bool {
+	log.Debugf("Unmounting file system mounted at %v", mountPoint)
+	if err := mounter.Unmount(mountPoint); err != nil {
+		log.WithError(err).Warnf("Failed to unmount file system mounted at %v", mountPoint)
```

Comment:
It looks like this comment is on outdated code from a previous commit. We already moved to using `mount.CleanUpMountPoint` instead of `mounter.Unmount` at all locations.